admin_guide/upgrading_citus.rst

```diff
@@ -39,6 +39,10 @@ Major and Minor Version Upgrades
 
 Major and minor version upgrades follow the same steps, but be careful: major upgrades can make backward-incompatible changes in the Citus API. It is best to review the Citus `changelog <https://github.com/citusdata/citus/blob/master/CHANGELOG.md>`_ before a major upgrade and look for any changes which may cause problems for your application.
 
+.. note::
+
+   Starting at version 8.1, Citus encrypts inter-node communication by default. SSL settings are configured automatically during an upgrade, with no action needed on your part. Just be careful when adding a new node to a Citus cluster that does not yet use SSL. The :ref:`adding a worker <adding_worker_node>` section covers that situation.
```

Comment:
I didn't talk about how if `sslmode=prefer` at the time of upgrade then Citus won't make the policy more strict. It doesn't seem relevant to the main warning, which is adding a new node to a pre-8.1 cluster.

Reply:
Another thing to think about is when someone upgrades all nodes of an 8.0 cluster to 8.1, then during the process the upgraded workers will be unable to talk to the un-upgraded ones. There will be a weird situation temporarily. Is this something we should cover, or is it as big of a problem as I am imagining?

Reply:
This statement is incorrect.

During an upgrade of an older cluster to 8.1 it specifically doesn't make changes to ssl. Only on new installations it enables internode encryption.

Reply:
Nils, specifically its this statement which is incorrect?: "SSL settings are configured automatically during an upgrade, with no action needed on your part"? 
The complexity only comes in if you add a *new* node to an existing cluster?

Reply:
I see what Nils means about an upgrade of an older cluster to 8.1 not making changes to ssl.

For the upgrade path, backend/distributed/citus--8.0-12--8.0-13.sql only calls citus_check_defaults_for_sslmode() when current_setting('ssl') is false. Furthermore, this function runs `ALTER SYSTEM SET citus.node_conninfo TO 'sslmode=prefer'` only if citus.node_conninfo is set to sslmode=require.

Just out of curiosity, under what situations will someone be doing an upgrade where ssl is false and sslmode=require? The name of citus_check_defaults_for_sslmode (and its comments) suggest that the default for citus.node_conninfo is sslmode=require, but in an upgrade that might not be true? The citus.node_conninfo GUC was first defined in v7.5, and in that version and in v8.0 it had a default value of sslmode=prefer. Starting in v8.1 it gets the default value sslmode=require when Citus is compiled with USE_SSL defined.

In any case, it looks like I should remove the statement about citus adjusting SSL during upgrade, but leave the warning link about adding a new 8.1 node to an upgraded cluster because the cluster may not have SSL enabled.

Reply:
Also odd that many `citus--8.0-*` scripts don't exist in v8.0.2 but appear in v8.1.0. I know pg doesn't care what the names of those files are, but it was confusing to see what I'm assuming is 8.1 functionality in an upgrade script named 8.0

Reply:
Sorry for the late reply

> Just out of curiosity, under what situations will someone be doing an upgrade where ssl is false and sslmode=require?

This happens when a user didn't change `citus.node_conninfo` on their deployment. In that case it is not store in any config file that gets loaded which causes postgres/citus to use the default value as hard coded in the application. Since we changed the hardcoded value it will change without the user noticing.

As you correctly noticed we change the value to the old default in this case, this will result in the user not observing a change to this guc when upgrading.